File: programmers/python 3/12921.py
# 문제 링크: https://programmers.co.kr/learn/courses/30/lessons/12921

def solution(n):
    primes = [False] * 2 + [True] * (n - 1)
    for i in range(2, int(n**.5) + 1):
        if primes[i]:
            for j in range(i**2, n + 1, i):
                primes[j] = False
    return len([p for p in primes if p])

File: programmers/swift/12921.swift
// 문제 링크: https://programmers.co.kr/learn/courses/30/lessons/12921

import Foundation

func solution(_ n: Int) -> Int {
    var primes = [Bool](repeating: true, count: n + 1)
    primes[..<2] = [false, false]

    for i in 2..<(Int(sqrt(Double(n))) + 1) {
        if !primes[i] { continue }

        for j in stride(from: Int(pow(Double(i), 2)), to: n + 1, by: i) {
            primes[j] = false
        }
    }

    return primes
        .filter { $0 }
        .count
}